components/gfx/text/glyph.rs

```diff
@@ -541,8 +547,41 @@ impl<'a> GlyphStore {
 
     #[inline]
     pub fn advance_for_char_range(&self, rang: &Range<CharIndex>) -> Au {
-        self.iter_glyphs_for_char_range(rang)
-            .fold(Au(0), |advance, (_, glyph)| advance + glyph.advance())
+        if !self.has_detailed_glyphs {
+            self.advance_for_char_range_simple_glyphs(rang)
+        } else {
+            self.iter_glyphs_for_char_range(rang)
+                .fold(Au(0), |advance, (_, glyph)| advance + glyph.advance())
+        }
+    }
+
+    #[inline]
+    fn advance_for_char_range_simple_glyphs(&self, rang: &Range<CharIndex>) -> Au {
+        let mask = u32x4::splat(GLYPH_ADVANCE_MASK);
+        let mut simd_advance = u32x4::splat(0);
+        let begin = rang.begin().to_usize();
+        let len = rang.length().to_usize();
+        let num_simd_iterations = len / 4;
+        let leftover_entries = rang.end().to_usize() - (len - num_simd_iterations * 4);
+        let buf: &[u32] = unsafe { mem::transmute(self.entry_buffer.as_slice()) };
```

Comment:
Can you factor this unsafe call out into a separate method and add a comment to the entry buffer field noting that if it is changed you may need to change the method too?

The safer way would be to use something like the nue crate here, but let's not pull in that dependency yet.